src/core/DomainDecomposition.cpp

```diff
@@ -222,17 +222,19 @@ void DomainDecomposition::setAtomsContiguous(int start) {
   if(gatindex.size()<getNumberOfAtoms()) shuffledAtoms=1;
   // keep in unique only those atoms that are local
   for(unsigned i=0; i<actions.size(); i++) actions[i]->unique_local_needs_update=true;
-  unique.clear();
+  unique.clear(); forced_unique.clear();
 }
 
 void DomainDecomposition::shareAll() {
-  unique.clear(); int natoms = getNumberOfAtoms();
+  unique.clear(); forced_unique.clear(); int natoms = getNumberOfAtoms();
   if( dd && shuffledAtoms>0 ) {
     for(int i=0; i<natoms; ++i) if( g2l[i]>=0 ) unique.push_back( AtomNumber::index(i) );
   } else {
     unique.resize(natoms);
     for(int i=0; i<natoms; i++) unique[i]=AtomNumber::index(i);
   }
+  forced_unique.resize( unique.size() );
```

Comment:
Here maybe you can copy the full vector without a loop (`forced_unique= unique`). However, should irrelevant because you correctly resized the vector first